### PR TITLE
Move to contrib spack-stack on Jet

### PR DESCRIPTION
**Pull request by InnocentSouopgui-NOAA**
## DESCRIPTION OF CHANGES:
Problem: Following the failure of the storage lfs4 on Jet, spack stack moved to another storage space: Contrib. This makes all softwares relying on spack stack libraries on Jet to fail.

Solution: Update jet modulefiles to point to the new installation of spack stack.

## TESTS CONDUCTED: 
The changes affects only Jet as all modifications are localized in Jet modulefile.

- [x] Compile branch Jet  using Intel
- [x] Extract Initial conditions for Global Workflow cycled experiment on Jet.
- [x] Global Workflow at resolution
  - [X] C96/48
  - [X] C192/96
  - [X] C384/192
- [x] Run relevant consistency tests locally on Jet.

Optional test.
Cycle Global workflow at resolutions 96/48, 192/96 and 384/192

## DEPENDENCIES:
None

## DOCUMENTATION:
Not needed.

## ISSUE: 
If this PR is resolving or referencing one or more issues, in this repository or elsewhere, list them here. For example, "Fixes issue mentioned in #123" or "Related to bug in https://github.com/NOAA-EMC/other_repository/pull/63"

Fixes issue mentioned in #977 
Refs NOAA-EMC/global-workflow#2841

### CODE REVIEW

**Comment by InnocentSouopgui-NOAA** [comment]
@GeorgeGayno-NOAA Can you review this so that we move it forward? The PR on global workflow depends on this.

**Comment by GeorgeGayno-NOAA** [comment]
I think your branch in behind the 'develop' branch. Please merge in the latest updates.

File: fix/link_fixdirs.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ pwd=$(pwd -P)
 if [ $machine = "hera" ]; then
     FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix"
 elif [ $machine = "jet" ]; then
```
Comment (GeorgeGayno-NOAA) [inline fix/link_fixdirs.sh:49]: The Jet directory does not exist.

File: util/gdas_init/set_fixed_files.sh [hunk 2]
```diff
@@ -36,7 +36,7 @@ if [ "$machine" = 'hera' ] ; then
 elif [ "$machine" = 'wcoss2' ] ; then
   FIX_ORO_INPUT=/lfs/h2/emc/global/noscrub/emc.global/FIX/fix/orog/20230615
 elif [ "$machine" = 'jet' ] ; then
-  FIX_ORO_INPUT=/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/orog/20230615
+  FIX_ORO_INPUT=/lfs5/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/orog/20230615
```
Comment (GeorgeGayno-NOAA) [inline util/gdas_init/set_fixed_files.sh]: Same issue here. This directory does not exist.

**Comment by InnocentSouopgui-NOAA** [comment]
> I think your branch in behind the 'develop' branch. Please merge in the latest updates.

Done

File: reg_tests/chgres_cube/driver.jet.sh [hunk 3]
```diff
@@ -54,7 +54,7 @@ fi
 
 export HOMEufs=$PWD/../..
 
-export HOMEreg=/lfs4/HFIP/hfv3gfs/emc.nemspara/role.ufsutils/ufs_utils/reg_tests/chgres_cube
+export HOMEreg=/lfs5/HFIP/hfv3gfs/emc.nemspara/role.ufsutils/ufs_utils/reg_tests/chgres_cube
```
Comment (GeorgeGayno-NOAA) [inline reg_tests/chgres_cube/driver.jet.sh:57]: This directory does not exist yet (I am creating them now). Did you run the consistency tests?

**Comment by GeorgeGayno-NOAA** [comment]
To get the `ice_blend` consistency test to run, I made these changes to the driver script (load the grib-util module, then remove the definitions of COPYGB, COPYGB2, and CNVGRIB).

```
diff --git a/reg_tests/ice_blend/driver.jet.sh b/reg_tests/ice_blend/driver.jet.sh
index 721b6bb6..2746d759 100755
--- a/reg_tests/ice_blend/driver.jet.sh
+++ b/reg_tests/ice_blend/driver.jet.sh
@@ -32,6 +32,7 @@ source ../../sorc/machine-setup.sh > /dev/null 2>&1
 module use ../../modulefiles
 module load build.$target.intel
 module load wgrib2/2.0.8
+module load grib-util/1.3.0
 set +x
 module list
 set -x
@@ -51,9 +52,6 @@ if [ "$UPDATE_BASELINE" = "TRUE" ]; then
 fi

 export WGRIB=/apps/wgrib/1.8.1.0b/bin/wgrib
-export COPYGB=/lfs5/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/grib-util-1.3.0-hrqavdi/bin/copygb
-export COPYGB2=/lfs5/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/grib-util-1.3.0-hrqavdi/bin/copygb2
-export CNVGRIB=/mnt/lfs5/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/grib-util-1.3.0-hrqavdi/bin/cnvgrib
```

**Comment by InnocentSouopgui-NOAA** [comment]
@GeorgeGayno-NOAA thanks for catching that.
It should be `/contrib/spack-stack/spack-stack-1.6.0` for the path to spack-stack. I completely missed that because I did not run it.

While we are at it, what are all the consistencies tests to run?
I caught everything that is used by global workflow and the making of ICS for global workflow.

Are there regression tests that can catch problems in most components of UFS_UTILS?

I need to do another round of checks/modification.

File: reg_tests/cpld_gridgen/rt.sh [hunk 4]
```diff
@@ -162,9 +162,9 @@ elif [[ $target = hercules ]]; then
     ulimit -s unlimited
     SBATCH_COMMAND="./cpld_gridgen.sh"
 elif [[ $target = jet ]]; then
-    STMP=${STMP:-/lfs4/HFIP/h-nems/$USER}
-    export MOM6_FIXDIR=/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/mom6/20220805
-    BASELINE_ROOT=/lfs4/HFIP/hfv3gfs/emc.nemspara/role.ufsutils/ufs_utils/reg_tests/cpld_gridgen/baseline_data
+    STMP=${STMP:-/lfs5/HFIP/h-nems/$USER}
+    export MOM6_FIXDIR=/lfs5/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/mom6/20220805
```
Comment (GeorgeGayno-NOAA) [inline reg_tests/cpld_gridgen/rt.sh]: MOM6_FIXDIR should be `/lfs5/HFIP/hfv3gfs/glopara/FIX/fix/mom6/20220805`

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA thanks for catching that. It should be `/contrib/spack-stack/spack-stack-1.6.0` for the path to spack-stack. I completely missed that because I did not run it.
> 
> While we are at it, what are all the consistencies tests to run? I caught everything that is used by global workflow and the makin of ICS for global workflow.
> 
> Are there regression tests that can catch problems in most components of UFS_UTILS?
> 
> I need to do another round of checks/modification.

The consistency/regression tests are in ./reg_tests. Most tests have a 'jet' driver script. Other tests have a single script that works for all machines. For instructions on how to run, see each scripts prolog. I am running the tests now. I recommend you run them as well. Let me know if you have questions.

File: reg_tests/ice_blend/driver.jet.sh [hunk 5]
```diff
@@ -51,11 +51,11 @@ if [ "$UPDATE_BASELINE" = "TRUE" ]; then
 fi
 
 export WGRIB=/apps/wgrib/1.8.1.0b/bin/wgrib
-export COPYGB=/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/grib-util-1.3.0-hrqavdi/bin/copygb
-export COPYGB2=/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/grib-util-1.3.0-hrqavdi/bin/copygb2
-export CNVGRIB=/mnt/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/grib-util-1.3.0-hrqavdi/bin/cnvgrib
+export COPYGB=/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/grib-util-1.3.0-74mdurc/bin/copygb
```
Comment (GeorgeGayno-NOAA) [inline reg_tests/ice_blend/driver.jet.sh:54]: You can remove the export of COPYGB, COPYGB2 and CNVGRIB if you add `module load grib-util/1.3.0` near the top of the script.

File: fix/link_fixdirs.sh [hunk 6]
```diff
@@ -47,7 +47,7 @@ pwd=$(pwd -P)
...
-    FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix"
+    FIX_DIR="/lfs5/HFIP/hfv3gfs/glopara/git_lfs4/fv3gfs/fix"
```
Comment (GeorgeGayno-NOAA) [inline fix/link_fixdirs.sh:50]: @KateFriedman-NOAA - I see multiple copies of the 'fixed' data on Jet. Is this the correct directory?

**Comment by InnocentSouopgui-NOAA** [comment]
@GeorgeGayno-NOAA , when you get the chance, have another look at the changes.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

File: fix/link_fixdirs.sh (hunk 6 above)
Comment (KateFriedman-NOAA) [reply fix/link_fixdirs.sh:50]: No, that is a copy from `/lfs4`. The official fix path on Jet is now: `/lfs5/HFIP/hfv3gfs/glopara/FIX/fix`

File: util/gdas_init/set_fixed_files.sh (hunk 2 above)
Comment (KateFriedman-NOAA) [reply util/gdas_init/set_fixed_files.sh]: This should be `/lfs5/HFIP/hfv3gfs/glopara/FIX/fix/orog/20230615`.

**Pull request merged**